packages/cli-e2e/__tests__/orgConfig.specs.ts

```diff
@@ -0,0 +1,158 @@
+import {listExtensions, listFields} from '../utils/platform';
+import {ensureDirSync} from 'fs-extra';
+import {foldersContainSimilarFiles} from '../utils/file';
+config({path: getPathToHomedirEnvFile()});
+
+describe('org:config', () => {
+  const testOrgId = process.env.TEST_ORG_ID!;
+  const {accessToken} = getConfig();
+  const snapshotProjectPath = join('snapshot-project');
+  const defaultTimeout = 10 * 60e3;
+  let processManager: ProcessManager;
+
+  const createNewTerminal = (
+    args: string[],
+    procManager: ProcessManager,
+    cwd: string,
+    debugName: string
+  ) => {
+    if (process.platform === 'win32') {
+      args.unshift('node');
+    }
+    return new Terminal(args.shift()!, args, {cwd}, procManager, debugName);
+  };
+
+  const previewChange = (targetOrg: string, procManager: ProcessManager) => {
+    const args: string[] = [
+      CLI_EXEC_PATH,
+      'org:config:preview',
+      `-t=${targetOrg}`,
+    ];
+
+    return createNewTerminal(
+      args,
+      procManager,
+      snapshotProjectPath,
+      'org-config-preview'
+    );
+  };
+
+  const pushToOrg = async (targetOrg: string, procManager: ProcessManager) => {
+    const args: string[] = [
+      CLI_EXEC_PATH,
+      'org:config:push',
+      '--skipPreview',
+      `-t=${targetOrg}`,
+    ];
+    const pushTerminal = createNewTerminal(
+      args,
+      procManager,
+      snapshotProjectPath,
+      'org-config-push'
+    );
+
+    await pushTerminal.when('exit').on('process').do().once();
+  };
+
+  const pullFromOrg = async (
+    targetOrg: string,
+    procManager: ProcessManager,
+    destinationPath: string
+  ) => {
+    const args: string[] = [
+      CLI_EXEC_PATH,
+      'org:config:pull',
+      `-t=${targetOrg}`,
+    ];
+
+    const pullTerminal = createNewTerminal(
+      args,
+      procManager,
+      destinationPath,
+      'org-config-pull'
+    );
+
+    await pullTerminal.when('exit').on('process').do().once();
+  };
+
+  beforeAll(async () => {
+    processManager = new ProcessManager();
+  });
+
+  afterAll(async () => {
+    await processManager.killAllProcesses();
+  });
+
+  describe('org:config:preview', () => {
+    it(
+      'should preview the snapshot',
+      async () => {
+        const previewTerminal = previewChange(testOrgId, processManager);
+
+        const expectedOutput = [
+          'Extensions',
+          '\\+   1 to create',
+          'Fields',
+          '\\+   2 to create',
+          'Filters',
+          '\\+   1 to create',
+        ].join('\\s*');
+        const regex = new RegExp(expectedOutput, 'gm');
+
+        await previewTerminal.when(regex).on('stdout').do().once();
+      },
+      defaultTimeout
+    );
+  });
+
+  describe('org:config:push', () => {
+    beforeAll(async () => {
+      await pushToOrg(testOrgId, processManager);
+    }, defaultTimeout);
+
+    it('should have pushed fields', async () => {
+      const fields: unknown[] = (await listFields(testOrgId, accessToken))
```

Comment:
I'm pretty sure we could have better typings than `unknown []` here. 
A list of fields is a known type, that you could probably import from platform-client.